.github/workflows/push_ci_docker_base.yml

```
@@ -21,4 +21,22 @@ jobs:
           echo ${{ secrets.GITHUB_TOKEN }} | docker login ghcr.io -u $GITHUB_ACTOR --password-stdin
           docker build . --tag ghcr.io/pytorch/cilint-clang-tidy:$GITHUB_SHA -f ci-docker-base/Dockerfile.cilint-clang-tidy
           docker push ghcr.io/pytorch/cilint-clang-tidy:$GITHUB_SHA
-
+      - name: Extract binary
+        run: |
+          set -ex
+          # Copying files directly from a docker image is not supported
+          # As a workaround, we create a temporary container, copy the binary, and remove it
+          image_id=$(docker create ghcr.io/pytorch/cilint-clang-tidy:$GITHUB_SHA)
+          docker cp $image_id:/clang-tidy-checks/build/bin/clang-tidy ./clang-tidy
+          docker rm -v $image_id
+      - uses: driazati/upload-artifact-s3@7faed57ea7f289192a68d1af3eb6358785e21b86
+        if: github.event_name == 'push' && github.ref == 'refs/heads/master'
+        name: Publish binary
+        with:
+          name: clang-tidy
+          if-no-files-found: error
+          s3-prefix: linux64
+          s3-bucket: oss-clang-format
+          path: ./clang-tidy
```

I'm not sure how [s3 interprets `./` in prefixes](https://alexwlchan.net/2020/08/s3-keys-are-not-file-paths/) (the upload-artifact-s3 action uses [`path` directly](https://github.com/seemethere/upload-artifact-s3/compare/main...driazati:main#diff-519fa009c5415a88d003e807b2f95d82d6acdd9651c8352aa7f6f9f808c44a0fL64) to create the complete s3 name), but it's all the same so we should just make this be

```suggestion
          path: clang-tidy
```